File: 0009. Palindrome Number/0009-Palindrome-Number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        
        rev, temp = 0, x
        while temp > 0:
            rev = (rev * 10) + (temp % 10)
            temp //= 10
        
        return rev == x

File: 0009. Palindrome Number/0009-Palindrome-Number.java
class Solution {
    public int reverse(int y) {
        int rev = 0;
        while (y > 0){
            int com = y % 10;
            rev = rev * 10 + com;
            y = y / 10;
        }
        return rev;
    }
    public boolean isPalindrome(int x) {
        // if (x < 0) { return false; }
        int rev = reverse(x);
        return rev == x;
    }
}
